decomposed/extensions/MDDE_LDM/Profile/Entity/Stereotypes/mdde_BusinessRule/Stereotypes/mdde_BusinessRuleWithMapping/Stereotypes/mdde_AggregateBusinessRule/Event_Handlers/CanCreate.xml

```diff
@@ -0,0 +1,25 @@
+<o:EventHandlerTargetItem Id="7768AB67-15D8-4C68-8754-5697D2960D23">
+<a:ObjectID>7768AB67-15D8-4C68-8754-5697D2960D23</a:ObjectID>
+<a:Name>CanCreate</a:Name>
+<a:CreationDate>1726842611</a:CreationDate>
+<a:Creator>RichaneRamsoedit</a:Creator>
+<a:TargetCategory.Type>1</a:TargetCategory.Type>
+<a:HandlerScript>Function %CanCreate%(parent)
+   Dim model : Set model = parent.Model
+   
+   &#39;Foutafhandeling wanneer er geen model is
+   If model Is Nothing Then
+      MsgBox &quot;Het model object kon niet worden opgehaald&quot;
+      %CanCreate% = false
+      Exit function
+   End If
+   
+   Dim pivotAttribute : pivotAttribute = model.GetExtendedAttribute(&quot;mdde_feature_business_rule_aggregate&quot;, 1)
+   
+   If pivotAttribute = False Then
+      %CanCreate% = False
+   Else
+      %CanCreate% = True
+   End If

```

Comment:
If I am not mistaken the full body if this function could be:
```
Function %CanCreate%(parent)
  %CanCreate% = parent.Model.GetExtendedAttribute('mdde_Feature_BusinessRule_Aggregate')
End Function
```

Reply:
Altough now when I read the documentation on the CanCreate event I wonder in what situation the event will be fired without a parent. Since in the example they check whether the parent is Nothing:

![image](https://github.com/user-attachments/assets/ec304fb3-0dd1-46e6-a176-8534358b4ffa)

We should probably check whether the parent is Nothing, since they check it. But checking whether the Model of a parent is known still doesn't make sense to me.

So I would say if the parent is Nothing CanCreate should be false, otherwise the anwser should be in the extended attribute:

```
Function %CanCreate%(parent)
  'Initialize the return value with False.
  %CanCreate% = False
    ' If the parent is known return the value of the MDDE configuration extended attribute.
   If parent Is Not Nothing Then
     %CanCreate% = parent.Model.GetExtendedAttribute("mdde_Feature_BusinessRule_Aggregate")
   EndIf
End Function
```

I am keen on hearing your thoughts.

Reply:
@harmen-xb I get the refactoring but I feel like if you refactor the code all down to 2 or 3 lines, without any variables, it might be less readable/understandable and therefore less transferrable (overdraagbaar) for maybe new users that want to modify the code in the extension.

Reply:
Function %CanCreate%(parent)
  'Initialize the return value with False.
  %CanCreate% = False
    ' If the parent is known return the value of the MDDE configuration extended attribute.
   If parent Is Not Nothing Then
     %CanCreate% = parent.Model.GetExtendedAttribute("mdde_Feature_BusinessRule_Aggregate")
   EndIf
End Function

This code gives the following error in PowerDesigner: Runtimefout Microsoft VBScript De objectvariabele is niet ingesteld (0x800A005B) At line 6, character 4